### DISCUSSION TITLE
Potential Bug in RDKit: Handling of Implicit Hydrogen Atoms at Chiral Centers

### DISCUSSION THREAD

**Discussion by ywu** [rdkit/rdkit#7723] [Development]
I am encountering an issue with how RDKit handles implicit hydrogen atoms at chiral centers. Specifically, when working with the following SMILES string: "S(Cc1oc(C(=O)N[C@@]2C(=O)NC(=O)CC2)cc1)C", `chem.AddHs` appears to return an incorrect structure where a hydrogen atom is missing at the chiral center.

According to Daylight's documentation,`[C@@]` is a correct SMILES notation that includes an implicit hydrogen atom bonded to the chiral center. This hydrogen should be considered the first neighbor "from" the chiral center. It seems that RDKit might not be handling this correctly.

Is this a bug in RDKit, or does RDKit use a different interpretation of the SMILES syntax?

**Comment by greglandrum** [comment]
Hi @ywu, atoms in square brackets in SMILES do not have additional implicit Hs added to them, so the C that you would like to be chiral is actually a radical. If you want implicit Hs on an atom in square brackets, you need to specify that, so the correct version of the SMILES for your molecule is:
`S(Cc1oc(C(=O)N[C@@H]2C(=O)NC(=O)CC2)cc1)C`

**Reply by ywu** [reply]
Hi Greg,

Thank you for your response. I apologize if my questions about SMILES are basic. According to [Daylight's documentation](https://www.daylight.com/dayhtml/doc/theory/theory.smiles.html):

> If the central carbon is not the very first atom in the SMILES and has an implicit hydrogen attached (it can have at most one and still be chiral), the implicit hydrogen is taken to be the first neighbor atom of the three neighbors that follow a tetrahedral specification. If the central carbon is first in the SMILES, the implicit hydrogen is taken to be the "from" atom. Hydrogens may always be written explicitly (as [H]) in which case they are treated like any other atom. In each case, the implied order is exactly as written in SMILES.

This suggests that implicit hydrogens are allowed for chiral carbons. Do I miss something? Thanks.

**Reply by greglandrum** [reply]
Yes, they have implicit Hs, but you need to say how many are there (sorry, my original answer above was not carefully enough formulated, I will update it). Here's the section in the Daylight docs:
> Within brackets, any attached hydrogens and formal charges must always be specified.

**Reply by ywu** [reply]
Hi Greg, I believe the rule you cited:

>  Within brackets, any attached hydrogens and formal charges must always be specified.

is in the context of:

>  Atoms with valences other than "normal" and elements not in the "organic subset" must be described in brackets.

For chiral carbons, which are with "normal" valences, there can be only one implicit hydrogen. My understanding of Daylight's documentation is as follows (please correct me if I’m mistaken):

-  When a hydrogen is bonded to a chiral carbon, it can be either explicit or implicit in the SMILES string.
-  If the hydrogen is explicit, it must be written in the "correct" order so that the order of the neighbors around the chiral center is consistent with the chirality.
-  If the hydrogen is implicit, it will be treated as the first neighbor after (to the right of) the chiral center if the chiral center is not the first atom in the SMILES string, or before (to the left of) the chiral center if the chiral center is the first atom in the SMILES string.

**Reply by greglandrum** [reply]
I am very sure that you have misread the documentation. If you want an atom in square brackets to have an implicit H added to it, you *must* put that H inside the square brackets. It makes no difference if that atom is from the organic set or not, this is determined by the square brackets (otherwise there would be no way to express species which have radicals).

To the second part: if the H is present in the square bracket, it is treated as if it immediately follows the C in the chiral order.
So `C[C@H](F)Cl` and `C[C@]([H])(F)Cl` correspond to exactly the same structure.

**Reply by ywu** [reply]
Thank you for the clarifications. After reading [the original SMILES paper](https://www.cs.tufts.edu/comp/150CSB/refs/1987%20%20SMILES,%20a%20chemical%20language%20and%20information%20system.%201.%20Introduction%20to%20methodology%20and%20encoding%20rules.pdf), I’m convinced that you were right all along. The paper says:

> Attached hydrogens and formal charges are always specified inside brackets. The number of attached hydrogens is shown by the symbol H followed by an optional digit. Similarly, a formal charge is shown by one of the symbols + or -, followed by an optional digit. If unspecified, the number of attached hydrogens and charges is assumed to be zero for an atom inside the bracket.
